### PR TITLE
Implement rules for basic spec

### PR DESCRIPTION
**Pull request by yanns**

### CODE REVIEW

**Comment by yanns** [comment]
@jrudolph do you review and merge PRs? Should I ping you on PRs?

**Comment by jrudolph** [comment]
Good point, pinging won't hurt :)

File: parboiled-core/src/main/scala-3/org/parboiled2/ParserMacros.scala [hunk 1]
```diff
@@ -239,7 +275,7 @@ class OpTreeContext(parser: Expr[Parser])(using Quotes) {
 
   private def reportError(error: String, expr: Expr[Any]): Nothing = {
     quotes.reflect.report.error(error, expr)
-    throw new Exception(error)
+    throw new scala.quoted.runtime.StopMacroExpansion
```
Comment (jrudolph) [inline parboiled-core/src/main/scala-3/org/parboiled2/ParserMacros.scala:278]: :+1: nice

**Review by jrudolph — APPROVED** [review]
LGTM

**Pull request merged**